ArchiSteamFarm/Statistics.cs

```diff
@@ -396,7 +396,7 @@ internal sealed class Statistics : IDisposable {
 
 				HashSet<(uint RealAppID, Steam.Asset.EType Type, Steam.Asset.ERarity Rarity)> skippedSetsThisUser = new HashSet<(uint RealAppID, Steam.Asset.EType Type, Steam.Asset.ERarity Rarity)>();
 
-				Dictionary<(uint RealAppID, Steam.Asset.EType Type, Steam.Asset.ERarity Rarity), Dictionary<ulong, uint>> theirTradableState = Trading.GetInventoryState(theirInventory);
+				(Dictionary<(uint RealAppID, Steam.Asset.EType Type, Steam.Asset.ERarity Rarity), Dictionary<ulong, uint>> theirFullState, Dictionary<(uint RealAppID, Steam.Asset.EType Type, Steam.Asset.ERarity Rarity), Dictionary<ulong, uint>> theirTradableState) = Trading.GetDividedInventoryState(theirInventory);
```

Comment:
I'm confused why you changed inventory state to divided inventory state here. I don't think it serves any purpose in the current code?

We're interested in checking only against tradable matches (as we can't trade other ones), so we're interested in `theirTradableState` like before. Since now the inventory includes both tradable and non-tradable items, you want something along `Trading.GetTradableInventoryState()`, you'll likely need to refactor `Trading` if it doesn't exist yet.

`theirFullState` seems useless for you since you depend on `IsNeutralOrBetter()` checking anyway, so the only work that you should do is generating all potential exchanges of our tradable items and their tradable items, and then checking which matches are neutral or better. I checked the current code in order to see if I've missed any logic bit but no, you iterate over `theirFullItems` derived from `theirFullState`, then you check against `theirTradableItems` derived from `theirTradableState` and continue if there are no tradables, which works, but is not optimized, you could iterate over tradable items to begin with, as it's always a subset of all.

Correct me if I'm wrong, otherwise consider my suggestion.

Reply:
You're right, I've used the function because it was already there. And since I already had the full state then I've checked if they needed our item in the first place to skip the following logic loop.

You're also right that we don't need that since `IsNeutralOrBetter()` checks that anyway. Will remove `theirFullState`.

Reply:
The check whether they need our item is nice but it's too trivial and doesn't achieve anything of value, it'd be OK if we needed that full state for something else anyway and we could optimize the remaining portion on that, but adding overhead and memory pressure for that alone isn't worth it. You can simply copy the current code from divided inventory state and get rid of parts you're not interested in.